phpBB/phpbb/db/migration/tool/permission.php

```diff
@@ -240,6 +240,25 @@ public function remove($auth_option, $global = true)
 		$this->auth->acl_clear_prefetch();
 	}
 
+	/**
+	 * Check if a permission role exists
+	 *
+	 * @param string $role_name The role name
+	 *
+	 * @return int The id of the role if it exists, 0 otherwise
+	 */
+	public function role_exists($role_name)
+	{
+		$sql = 'SELECT role_id
+			FROM ' . ACL_ROLES_TABLE . "
+			WHERE role_name = '" . $this->db->sql_escape($role_name) . "'";
+		$result = $this->db->sql_query($sql);
```

Comment:
`$this->db->sql_query_limit($sql, 1);` maybe?

On a side note I am used to let the function return the bool, 

`return true if the role exists, false otherwise.`

Reply:
If you look at the changes we actually need the role ID in some cases. No need to return a Boolean.

Reply:
Ah ok, I use a similar _(quite the same)_ query for the extensions, that's why I asked.
Wouldn't it be better, codewise, to limit that query though?

Reply:
No. There will only be 1 result ever. And the roles table is VERY small. By default only 24 items. There would be no measurable performance gain. See (both same query time):
![Screen Shot 2020-06-14 at 11 48 26 PM](https://user-images.githubusercontent.com/303711/84626228-9918f380-ae99-11ea-89f7-a8a318adb527.png)